apiary-metastore-events/apiary-metastore-consumer/privileges-grantor/privileges-grantor-core/src/main/java/com/expediagroup/apiary/extensions/events/metastore/consumer/privilegesgrantor/core/Grantor.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (C) 2018-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.apiary.extensions.events.metastore.consumer.privilegesgrantor.core;
+
+public enum Grantor {
```

Comment:
Just to understand this better, whats the purpose of this Grantor ("admin") ?

Is it the name for who is granting the privilege ?

Reply:
If its the name of the person granting the privilege, then I think this should be changed to better reflect who is granting the priviliege. Perhaps, you can put name of this module "apiary-privilege-grantor".

This will better reflect who granted this privilege on the table.

Reply:
yes, `Grantor` is a user type granting the privilege. I followed the `hive-metastore` [api](http://hive.apache.org/javadocs/r1.2.2/api/org/apache/hadoop/hive/metastore/api/PrivilegeGrantInfo.html#PrivilegeGrantInfo(java.lang.String,%20int,%20java.lang.String,%20org.apache.hadoop.hive.metastore.api.PrincipalType,%20boolean)) constructor parameter name.  

I could use the name `PrivilegeGrantor` but then it would conflict with the class `PrivilegesGrantor` which contains the actual logic for granting the privileges.

Reply:
Alright. The argument type is a string. So, I think you can put `apiary-privilege-grantor` in there to tell that a particular privilege on a table was granted by this module. What do others think ? @patduin @massdosage

Reply:
I think I misunderstood the message, originally, I thought you suggested the rename of the `Grantor` class name, but the suggestion is to pass `apiary-privilege-grantor` string that we pass when initializing `PrivilegeGrantInfo`.  
In this case, if `client.grant_privileges(privilegeBag);` method does not do any downstream checks for Grantor types, it can be updated from `Admin` -> `apiary-privilege-grantor`

Reply:
^^ Yeah. Thats what I was suggesting. Apologies for not being very clear.

Reply:
Updated, thanks!

![image](https://user-images.githubusercontent.com/2702432/57781049-4febf280-7721-11e9-814d-f3be86f9f876.png)

Reply:
Perfect ! thats what I was looking for 🎉